src/main/scala/net/snowflake/spark/snowflake/SnowflakeTelemetry.scala

```diff
@@ -245,13 +245,15 @@ object SnowflakeTelemetry {
       logger.warn(s"Fail to send OOB Telemetry message: ${th.getMessage}")
   }
 
-  def send(telemetry: Telemetry): Unit = {
+  def send(telemetry: Telemetry): Unit = if (telemetry != null) {
     var curLogs: List[(ObjectNode, Long)] = Nil
     this.synchronized {
       curLogs = logs
       logs = Nil
     }
     telemetryMessageSender.send(telemetry, curLogs)
+  } else {
+    logger.warn("Telemetry message not sent because Telemetry client is null")
```

Comment:
This warn will be prompted for every telemetry message. Should just check one time when initializing telemetry service?

Reply:
```
lazy val skipTelemetry = {
   val skip = telemetry != null
   if (skip) warn
   skip
}
if (! skipTelemetry) {
....
}
```

Reply:
Changed like below:
```
  private lazy val telemetryClientNullWarningSent = {
    logger.warn("Telemetry message not sent because Telemetry client is null")
    true
  }

  def send(telemetry: Telemetry): Unit = if (telemetry != null) {
    var curLogs: List[(ObjectNode, Long)] = Nil
    this.synchronized {
      curLogs = logs
      logs = Nil
    }
    telemetryMessageSender.send(telemetry, curLogs)
  } else {
    telemetryClientNullWarningSent
  }
```